protocol/x/assets/keeper/asset.go

```diff
@@ -16,24 +17,34 @@ import (
 
 func (k Keeper) CreateAsset(
 	ctx sdk.Context,
+	assetId uint32,
 	symbol string,
 	denom string,
 	denomExponent int32,
 	hasMarket bool,
 	marketId uint32,
 	atomicResolution int32,
 ) (types.Asset, error) {
-	// Get the `nextId`
-	nextId := k.GetNumAssets(ctx)
+	if prevAsset, exists := k.GetAsset(ctx, assetId); exists {
+		return types.Asset{}, errorsmod.Wrapf(
+			types.ErrAssetIdAlreadyExists,
+			"previous asset = %v",
+			prevAsset,
+		)
+	}
 
-	_, found := k.internalGetIdByDenom(ctx, denom)
-	if found {
+	if _, exists := k.internalGetIdByDenom(ctx, denom); exists {
 		return types.Asset{}, errorsmod.Wrap(types.ErrAssetDenomAlreadyExists, denom)
 	}
 
+	// Ensure assetId zero is always USDC. This is a protocol-wide invariant.
+	if assetId == types.AssetUsdc.Id && denom != types.AssetUsdc.Denom {
```

Comment:
should we add the other validation? i.e. if denom == usdc, then id must be 0:

there's prob a shorter way to write this in 1 single if statement (some combo of XOR maybe)
```
if assetid == assetUsdc.Id || denom == assetUsdc.denom {
    if assetid != assetUsdc.Id || denom != assetUsdc.denom {
        throw error
    }
}
```

Reply:
I just added a separate condition - feels cleaner that way